optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/DefaultSolver.java

```diff
@@ -125,17 +126,17 @@ public String explainBestScore() {
 
     @Override
     public long getTimeMillisSpent() {
-        Long startingSystemTimeMillis = solverScope.getStartingSystemTimeMillis();
-        if (startingSystemTimeMillis == null) {
+        Long startingSystemTimeNanos = solverScope.getStartingSystemTimeNanos();
+        if (startingSystemTimeNanos == null) {
             // The solver hasn't started yet
             return 0L;
         }
-        Long endingSystemTimeMillis = solverScope.getEndingSystemTimeMillis();
-        if (endingSystemTimeMillis == null) {
+        Long endingSystemTimeNanos = solverScope.getEndingSystemTimeNanos();
+        if (endingSystemTimeNanos == null) {
             // The solver hasn't ended yet
-            endingSystemTimeMillis = System.currentTimeMillis();
+            endingSystemTimeNanos = System.nanoTime();
```

Comment:
This is dangerous, from https://www.geeksforgeeks.org/java-system-nanotime-vs-system-currenttimemillis/

Cons:

    The result reflected doesn’t have any fixed reference point. According to Java documentation,

    The value returned represents nanoseconds since some fixed
    but arbitrary time (perhaps in the future, so values may be negative).

    Less accurate. This method provides nanosecond precision, but not necessarily nanosecond accuracy. No guarantees are made about how frequently values change.
    Depending on the system, it can take more than 100 cpu cycles to execute.
    Not thread safe. May return erroneous results if used between more than one threads.

Reply:
Also, a Long can hold about 4 000 000 000 000 000 000 nanos.
So 4 000 000 000 seconds. (think milli, micro, nano)
So 46296 days. So 126 years. This means OptaPlanner might have a Y2K bug after 2096 (the epoch starts in 1970).
Ok, this is less important, but the previous comment highlights some risks. I don't think this change should be part of this PR, it should be in a seperate PR.